Neos.Flow/Classes/Session/Http/SessionRequestComponent.php

```diff
@@ -0,0 +1,86 @@
+<?php
+namespace Neos\Flow\Session\Http;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Http\Component\ComponentContext;
+use Neos\Flow\Http\Component\ComponentInterface;
+use Neos\Flow\Http\Cookie;
+use Neos\Flow\Session\SessionManager;
+use Neos\Flow\Session\SessionManagerInterface;
+use Neos\Flow\Utility\Algorithms;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * A HTTP component that initialises the standard Flow session with cookie information.
+ */
+class SessionRequestComponent implements ComponentInterface
+{
+    /**
+     * @Flow\InjectConfiguration(package="Neos.Flow", path="session")
+     * @var array
+     */
+    protected $sessionSettings;
+
+    /**
+     * @Flow\Inject(lazy=false)
+     * @var SessionManagerInterface
+     */
+    protected $sessionManager;
+
+    /**
+     * @param ComponentContext $componentContext
+     */
+    public function handle(ComponentContext $componentContext)
+    {
+        if (!$this->sessionManager instanceof SessionManager) {
```

Comment:
This probably hints at a deficit in the SessionManagerInterface, but out of scope of this PR.

Reply:
Yep it does and correct. IMHO we should actually set the interface to internal or remove it, but that's for another change.